modules/tensor_mechanics/src/userobjects/ElementPropertyReadFile.C

```diff
@@ -24,11 +24,11 @@ ElementPropertyReadFile::validParams()
   params.addParam<FileName>("prop_file_name", "", "Name of the property file name");
   params.addRequiredParam<unsigned int>("nprop", "Number of tabulated property values");
   params.addParam<unsigned int>("ngrain", 0, "Number of grains");
-  params.addRequiredParam<MooseEnum>(
-      "read_type",
-      MooseEnum("element grain"),
-      "Type of property distribution: element:element by element property "
-      "variation; grain:voronoi grain structure");
+  params.addParam<unsigned int>("nblock", 0, "Number of blocks");
+  params.addParam<MooseEnum>("read_type",
+                             MooseEnum("element grain block none", "none"),
```

Comment:
Drop the none option and do not specify a default (second parameter). That way the parameter becomes required and you do not have to error check yourself.